### ISSUE TITLE
Set up a new communication channel for the community 

### ISSUE THREAD

**Issue by gingerkidney** [taichi-dev/taichi#4878]
TL;DR: Set up a new Slack Workspace for the communication.

Two weeks ago I started a discussion about the communication platform in slack #github-community. Thanks to @quadpixels @k-ye and fengliang joined the discussion. 

Currently, Taichi community is using slack for communication. But it's not friendly because users need to send an email to us and then get invited. They can not access the channel by a direct invitation link. A separate workspace should work, just like many other open-source projects do to solve the problem. 

Other tools like zulip and discord are mentioned to have a similar function as slack in the discussion. These platforms are widely used in the community also. But it's hard to maintain channels on different platforms. Then we thought about [Matrix](https://matrix.org/). Matrix is an open-source project that publishes the Matrix open standard for secure, decentralized, real-time communication, and its Apache licensed reference implementations. It supports integrating with different platforms through its API. So if someone uses discord and wants to join Taichi community, they can help to bridge the slack channel with discord. It's convenient for those who do not want to chat on different platforms. We will help with the slack authorization if needed. 

So we will keep using Slack as our main communication. And please join the new workspace [here](https://join.slack.com/t/taichicommunity/shared_invite/zt-14ic8j6no-Fd~wKNpfskXLfqDr58Tddg). The new invitation link will be updated on readme and communication.

**Comment by ailzhang** [comment]
@gingerkidney would you mind also clarifying who are we targeting for this new communication channel? 
I'm mainly curious what kind of discussion do we expect to happen on slack compared to the current github/forum? I think it might help our community members to decide whether they should to join slack and stay active on it ;) 
For example, are taichi users or developers the target users? Do we want users to ask questions in slack (we might prefer forum and github discussion for these indeed)?

**Comment by gingerkidney** [comment]
> @gingerkidney would you mind also clarifying who are we targeting for this new communication channel? I'm mainly curious what kind of discussion do we expect to happen on slack compared to the current github/forum? I think it might help our community members to decide whether they should to join slack and stay active on it ;) For example, are taichi users or developers the target users? Do we want users to ask questions in slack (we might prefer forum and github discussion for these indeed)?

Good point! Taichi is using GitHub discussion as a forum. Any question related to using Taichi is welcome. Slack is instant communication and more development-oriented. Discussion about the internals, RFC, and new features implementation is expected. Release notes and events will be published in the slack also.
